firmware/Core/Src/telecommands/telecommand_parser.c

```diff
@@ -270,13 +326,49 @@ uint8_t TCMD_parse_full_telecommand(const char tcmd_str[], TCMD_TelecommandChann
         }
     }
 
-    // TODO: read out the @sha256=xx hash and validate it here.
+    // Extract @sha256=xxxx from the telecommand string, starting at &tcmd_str[end_of_args_idx]
+    if (SHA256_ENABLED) {
+        if (GEN_get_index_of_substring_in_array(tcmd_suffix_tag_str, tcmd_suffix_tag_str_len, "@sha256=") < 0) { 
+            // Failure: sha256 suffix tag not found.
+            LOG_message(
+                LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                "Error: TCMD_parse_full_telecommand: sha256 validation suffix tag not found in telecommand string."
+            );
+            return 90;
+        }
+
+        uint8_t parsed_sha256_hash[32];
+        // Parse the sha256 hash from the suffix tag.
+        if (TCMD_get_suffix_tag_hex_array(tcmd_suffix_tag_str, "@sha256=", parsed_sha256_hash) != 0) {
+            LOG_message(
+                LOG_SYSTEM_TELECOMMAND, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                "Error: TCMD_parse_full_telecommand: failed to parse present @sha256=xxxx."
+            );
+            return 100;
+        }
+
+        // generate a string that contains 'CTS1+<tcmd_name>(<params>)'
+        uint8_t tcmd_name_with_parens[end_of_args_idx + 1];
```

Comment:
Instead of copying this to a new string, couldn't you just call `CRYPT_compute_sha256_hash` like this:

```c
CRYPT_compute_sha256_hash(
    tcmd_str, end_of_args_idx + 1
);